src/ddscxx/include/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp

```diff
@@ -312,6 +613,42 @@ void write_string(basic_cdr_stream& str, const T& towrite, size_t N)
   str.alignment(1);
 }
 
+/**
+ * @brief
+ * Bounded string write function, byteswapped version.
+ *
+ * Same as write_string(), only that the string length is byte swapped.
+ *
+ * @param[in, out] str The stream to write to.
+ * @param[in] towrite The string to write.
+ * @param[in] N The maximum number of characters to write to the stream.
+ */
+template<typename T>
+void write_string_swapped(basic_cdr_stream& str, const T& towrite, size_t N)
+{
+  if (str.abort_status())
+    return;
+
+  size_t string_length = towrite.length();
+  if (N &&
+      string_length > N) {
+    if (str.status(serialization_status::write_bound_exceeded))
+      return;
+  }
+
+  //add 1 extra for terminating NULL
+  string_length++;
```

Comment:
This invalidates the previous boundary check. It's unlikely it'll happen, but still.